docs: `*_invitation` life cycle clarification (#756)
mongodb/terraform-provider-mongodbatlas: website/docs/r/project_invitation.html.markdown

-> **NOTE:** Groups and projects are synonymous terms. You may find GROUP-ID in the official documentation.

> ~> **IMPORTANT:** This resource is specifically to mange the invitations, not the user beyond that. Possible provider behavior depending on the invitee's action:

Reviewer: ```suggestion
~> **IMPORTANT:** This resource is only for managing invitations, not for managing the Atlas User being invited. Possible provider behavior depending on the invitee's action:
```
